Autopilot/PathManager/Src/CommWithAttitudeManager.cpp

```diff
@@ -32,27 +37,28 @@ void SendCommandsForAM(CommandsForAM *commands)
     *commandsOut = *commands;
 
     //Post mail slot to mail queue
-    osMailPut(commandsMailQ, commandsOut);
+    osStatus sendStatus = osMailPut(commandsMailQ, commandsOut);
 }
 
-bool GetAttitudeData(AttitudeData *data)
+
+bool GetFromPMToAM(CommandsForAM *commands)
 {
-    //Try to get data from mail queue
+    //Try to get commands from mail queue
     osEvent event;
-    AttitudeData * dataIn;
-    event = osMailGet(attitudeDataMailQ, 0);
+    CommandsForAM * commandsIn;
```

Comment:
Your'e both reaing and writing commandsForAm. That's an issue. You should be writing commsWithAM, so that attitude manager can collect adn reading AttitudeData, so path manager can get attitude manager's data.

Reply:
The header file still reads AttitudeData. Just the definitions have been moved so that each cpp file only deals with a single queue. Maybe this file should be renamed something like CommFromPMToAM.cpp as it no longer corresponds solely to CommWithAttitudeManager.hpp but rather provides some definitions for both header files.